src/server.rs

```diff
@@ -365,10 +382,21 @@ async fn index(
     let mut request = Request::from_actix_request(&req, body, &global_request_headers);
 
     // Before middleware
-    request = match middleware_router.get_route(&MiddlewareRoute::BeforeRequest, req.uri().path()) {
+    // Global
+    request = middleware_router
+        .get_global_middlewares(&MiddlewareType::BeforeRequest)
+        .iter()
+        .fold(request, |request, function| {
+            block_on(execute_middleware_function(&request, function)).unwrap_or_else(|e| {
```

Comment:
@AntoineRR , I just realized. Wouldn't this bring about a performance downgrade?